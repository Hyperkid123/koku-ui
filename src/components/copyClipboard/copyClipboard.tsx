import { Button, TextInput } from '@patternfly/react-core';
import { OkIcon, PasteIcon } from '@patternfly/react-icons';
import React from 'react';
import copyfnc from './copyFunc';

interface ClipboardText {
  text: string;
  'aria-label': string;
}
interface CopyText {
  isCopied: boolean;
  onCopy: (event: any) => void;
}

export const CopyClipboardStateless: React.SFC<ClipboardText & CopyText> = ({
  text,
  isCopied,
  'aria-label': ariaLabel,
  onCopy,
}) => (
  <div style={{ position: 'relative' }}>
    {/* Update this once Input Group is merged:
        https://github.com/patternfly/patternfly-react/pull/1108 */}
    <TextInput
      onChange={() => null}
      isReadOnly
      value={text}
      aria-label={ariaLabel}
    />
    <Button
      onClick={onCopy}
      variant="primary"
      style={{ position: 'absolute', right: '1px' }}
    >
      {isCopied ? <OkIcon /> : <PasteIcon />}
    </Button>
  </div>
);

interface State {
  wasClicked: boolean;
}

export class CopyClipboard extends React.Component<ClipboardText, State> {
  constructor(props) {
    super(props);
    this.state = { wasClicked: false };
  }

  public render() {
    return (
      <CopyClipboardStateless
        text={this.props.text}
        isCopied={this.state.wasClicked}
        aria-label={this.props['aria-label']}
        onCopy={(event: React.FormEvent<HTMLButtonElement>) => {
          copyfnc(event, this.props.text);
          this.setState({ wasClicked: true }, () => {
            const timer = setTimeout(() => {
              this.setState({ wasClicked: false }, () => {
                clearTimeout(timer);
              });
            }, 2000);
          });
        }}
      />
    );
  }
}
