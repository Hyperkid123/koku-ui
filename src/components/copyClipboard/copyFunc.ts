import React from 'react';

const copy = (event: React.FormEvent<HTMLButtonElement>, text): void => {
  const clipboard = event.currentTarget.parentElement;
  const el = document.createElement('input');
  el.value = text;
  clipboard.appendChild(el);
  el.select();
  document.execCommand('copy');
  clipboard.removeChild(el);
};

export default copy;
