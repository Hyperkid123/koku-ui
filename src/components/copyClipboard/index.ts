export { CopyClipboard as default } from './copyClipboard';
