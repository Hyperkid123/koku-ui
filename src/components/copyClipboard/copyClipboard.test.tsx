import { OkIcon, PasteIcon } from '@patternfly/react-icons';
import { shallow } from 'enzyme';
import React from 'react';
import { CopyClipboard, CopyClipboardStateless } from './copyClipboard';

test('copy clipboard matches snapshot', () => {
  expect(
    shallow(
      <CopyClipboardStateless
        text="this is text to copy"
        onCopy={jest.fn}
        isCopied={false}
        aria-label="copy me"
      />
    )
  ).toMatchSnapshot();
});

test('copy clipboard different icon on isCopied', () => {
  let view = shallow(
    <CopyClipboardStateless
      text=""
      onCopy={jest.fn}
      isCopied={false}
      aria-label="copy text"
    />
  );
  expect(view.find(PasteIcon).length).toBe(1);
  expect(view.find(OkIcon).length).toBe(0);

  view = shallow(
    <CopyClipboardStateless
      text=""
      onCopy={jest.fn}
      isCopied
      aria-label="copy text test"
    />
  );
  expect(view.find(PasteIcon).length).toBe(0);
  expect(view.find(OkIcon).length).toBe(1);
});
