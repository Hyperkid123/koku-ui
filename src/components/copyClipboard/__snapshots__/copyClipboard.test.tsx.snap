// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`copy clipboard matches snapshot 1`] = `
<div
  style={
    Object {
      "position": "relative",
    }
  }
>
  <TextInput
    aria-label="copy me"
    className=""
    isDisabled={false}
    isReadOnly={true}
    isRequired={false}
    isValid={true}
    onChange={[Function]}
    style={null}
    type="text"
    value="this is text to copy"
  />
  <Button
    aria-label={null}
    className=""
    component="button"
    isActive={false}
    isBlock={false}
    isDisabled={false}
    isFocus={false}
    isHover={false}
    isInline={false}
    onClick={[Function]}
    style={
      Object {
        "position": "absolute",
        "right": "1px",
      }
    }
    type="button"
    variant="primary"
  >
    <PasteIcon
      color="currentColor"
      size="sm"
      title={null}
    />
  </Button>
</div>
`;
